# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1203973788 18000
#      Mon Feb 25 16:09:48 2008 -0500
# Node ID 7e1b042c5418240ec84cc3bf46e1f9d91d060105
# Parent  26d8a92644de6c84f6052e4766bf662cf553c97b
octave_scalar::round: use xround instead of ::round (MSVC does not have ::round).

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-02-25  Michael Goffioul  <michael.goffioul@gmail.com>
+
+	* ov-scalar.cc (octave_scalar::round): Use xround instead of ::round
+	in mapper implementation.
+
 2008-02-25  John W. Eaton  <jwe@octave.org>
 
 	* ov-base.cc (UNDEFINED_MAPPER, STRING_MAPPER): New macros.  Use
 	them to define mapper functions.
 
 	* mappers.cc (Fisalnum, Fisalpha, Fisascii, Fiscntrl, Fisdigit,
 	Fisgraph, Fislower, Fisprint, Fispunct, Fisspace, Fisupper,
 	Fisxdigit, Ftoascii, Ftolower, Ftoupper):
diff --git a/src/ov-scalar.cc b/src/ov-scalar.cc
--- a/src/ov-scalar.cc
+++ b/src/ov-scalar.cc
@@ -326,17 +326,17 @@ SCALAR_MAPPER (cos, ::cos)
 SCALAR_MAPPER (cosh, ::cosh)
 SCALAR_MAPPER (exp, ::exp)
 SCALAR_MAPPER (fix, ::fix)
 SCALAR_MAPPER (floor, ::floor)
 SCALAR_MAPPER (imag, ::imag)
 CD_SCALAR_MAPPER (log, ::log, std::log, 0.0, octave_Inf)
 CD_SCALAR_MAPPER (log10, ::log10, std::log10, 0.0, octave_Inf)
 SCALAR_MAPPER (real, ::real)
-SCALAR_MAPPER (round, ::round)
+SCALAR_MAPPER (round, xround)
 SCALAR_MAPPER (signum, ::signum)
 SCALAR_MAPPER (sin, ::sin)
 SCALAR_MAPPER (sinh, ::sinh)
 CD_SCALAR_MAPPER (sqrt, ::sqrt, std::sqrt, 0.0, octave_Inf)
 SCALAR_MAPPER (tan, ::tan)
 SCALAR_MAPPER (tanh, ::tanh)
 SCALAR_MAPPER (finite, xfinite)
 SCALAR_MAPPER (isinf, xisinf)
